civictechprojects/static/css/partials/_Profile.scss

```diff
@@ -114,6 +114,34 @@
   }
 }
 
+// for use with IframeGroupDisplay to make it full size in iframe
+.frame-full {
+  margin: 0;
+  width: 100%!important; // important to override the media queries
+}
+
+.Profile-primary-container.frame-full .AboutGroup-card-container .row .ProjectCardContainer .row{
+  margin: 0;
+  row-gap: 1rem;
+  column-gap: 1rem;
+  justify-content: space-evenly;
+  align-items: center;
+  height: 100vh; // force the flex box to use the whole iframe height that it is in
+}
+
+.Profile-primary-container.frame-full .AboutGroup-card-container .row .ProjectCardContainer .row .col-sm-12.col-lg-6 .ProjectCard-root {
+  margin: 0;
+}
+
+// for the project cards burried in the iframe, give them a fixed width and let them fill the frame
+.Profile-primary-container.frame-full .AboutGroup-card-container .row .ProjectCardContainer .row .col-sm-12.col-lg-6 {
+  flex: 0 0 26rem;
+}
+
+.use-parent-width {
+  max-width: none;
```

Comment:
This one is applying inconsistently. It works for a project page when that page is below the one-column breakpoint (for us, lg in bootstrap terms) but above that it leads to part of the page expanding and part not:
![image](https://github.com/DemocracyLab/CivicTechExchange/assets/16870466/510c3e8d-329e-4254-8319-6b9a6067e08e)

max-width should return to its prior behavior (controlled by the .container class) at that point -- try wrapping this rule in `@include media-breakpoint-down(lg) { ... }` and see if it works better? That should make it apply at lower breakpoints and disappear at higher ones, which means the page retains the designed widths correctly. The page layout/content has an upper bound for usability reasons and should be retained even in a framed view.

Reply:
So I think I went the other way from what you were thinking.  I'm letting the project fill the full with of the viewport.  Based on the logic that the parent page will be imposing it's own container width, and breakpoints.

The fix was just to let "Profile-primary-container" go to width:100% if in an iframe. 

<img width="1412" alt="image" src="https://github.com/DemocracyLab/CivicTechExchange/assets/3317487/5bde612a-b7dc-4d93-83e8-d5463dc2303a">

Reply:
I tested that too, but no, text that wide is basically unreadable. Optimal line length for reading comprehension -- look at any newspaper, they do columns of text and not one fullpage line. Or most news sites constrain article width, etc. 

More generally, this page was designed deliberately, by the dlab design team, and so I think the default should be not to change that. Please revert this change so it has the same maximum content width as if it were displayed natively.

Reply:
If you're interested, https://baymard.com/blog/line-length-readability is a good place to start.

Reply:
The problem is that with the max-width there, even within a normal width viewport, we get a "AboutProjectDisplay" with blank-space margins on the sides. It's because these max-width breakpoints were set for the project withing the rest of the DLab html page, but now we are trying to provide the project as a component to be embedded in another website.

<img width="571" alt="image" src="https://github.com/DemocracyLab/CivicTechExchange/assets/3317487/dce1eea1-7759-46fe-b33d-4e2b65121434">

The site where this is being embedded won't want that much space on the sides and will want to control the padding/margin around our component themselves. 

We need to push the AboutProjectDisplay all the way to the edges of the viewport. 

I fully agree that the 3000px wide viewport is not useable - but the most common use case is that the developer using our iframe will put it under their own container div that will have it's own max-width.  I'm not overriding the max-width because I want it to go crazy wide, it's just about the extra margin on the sides. 

Here is the opensandiego example:
https://github.com/opensandiego/opensandiego.github.io/blob/master/_layouts/home.html#L32-L61

See that they already have their own container class https://github.com/opensandiego/opensandiego.github.io/blob/master/css/main.scss#L70

We can put something in the instructions telling developers that this should be used within a container no wider than 1392px, though I think it looks fine when a little wider too.

Or I can add a max-width if we are in an iframe to the MainController if we want to make it an enforced requirement.

What should we do?

Reply:
Hm, okay, I see what you're getting at. I agree, we don't want that whitespace there for narrower iframes. I apologize if I suggested that, I don't want it either! Only at very wide iframe width do I want to see changes. Thanks for explaining - I think we're agreeing on the optimal outcome here, it's just an implementation question?

Specifically,
- iframed profile page content is 100% of iframe width up to a defined breakpoint
- at that breakpoint, profile page content stops growing in width (so it'll have left/right whitespace)
  - (optionally) we can tell developers the iframe should probably have a max-width declared


If that's correct ... Well, first, after reading your comments, I don't think this is a blocking issue for release. Marlon - good to go on this. As you say it's very likely to be an edge case due to containers on the embedding website.

But on that note, I'd like to solve this on dlab's side. Then users don't have to do anything special, they just embed the iframe and it "just works."

Do you think it's a problem if some hypothetical user wants to render the world's longest row of project cards at 3000px width or whatever? If you do, maybe MainController is the place to put that max-width; I'd defer to you in that case. But if you don't think cards having unbound width is an issue, I'd be happy to try and figure out a targeted solution for the profile page only.  Let me know what approach you'd prefer - I'd be happy with either.

Reply:
In a target solution for profiles, I figured out that I can just set max-width (in .use-parent-width) to 1392px instead of none.  And I will also change the name to .override-breakpoint-max-width.  So I don't need to do it in MainController.

But I will wait, unless you want this now.

Reply:
Let's wait; release testing's already started. I'm also thinking we may be able to approach this with Bootstrap's `.container-fluid` class, but that's definitely a wait-and-try-later level change.